exchange/auction.go

```diff
@@ -91,53 +91,56 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 	for _, topBidsPerImp := range a.winningBidsByBidder {
 		for _, topBidPerBidder := range topBidsPerImp {
 			impID := topBidPerBidder.bid.ImpID
-			var customCacheKey string
-			var catDur string
-			useCustomCacheKey := false
-			if competitiveExclusion && topBidPerBidder == a.winningBids[impID] {
-				// set custom cache key for winning bid when competitive exclusion applies
-				catDur = bidCategory[topBidPerBidder.bid.ID]
-				if len(catDur) > 0 {
-					customCacheKey = fmt.Sprintf("%s_%s", catDur, hbCacheID)
-					useCustomCacheKey = true
-				}
-			}
-			if bids {
-				if jsonBytes, err := json.Marshal(topBidPerBidder.bid); err == nil {
-					if useCustomCacheKey {
-						// not allowed if bids is true; log error and cache normally
-						errs = append(errs, errors.New("cannot use custom cache key for non-vast bids"))
+			isOverallWinner := a.winningBids[impID] == topBidPerBidder
+			if includeBidderKeys || (includeWinners && isOverallWinner) {
```

Comment:
You could simply test for the opposite in the above line and, if true, do a continue.  This would reduce the need for indenting and might make comparing the new and old code quite a bit easier since no additional indentation would be needed.  In other words, replace above line with:
if ! includeBidderKeys && ! isOverallWinner {
    continue
}

Reply:
That's probably more elegant. Let me change it

Reply:
changed. The new version of the code implements your suggestion Joe